projects/angular/src/data/datagrid/datagrid.ts

```diff
@@ -280,6 +280,28 @@ export class ClrDatagrid<T = any> implements AfterContentInit, AfterViewInit, On
           this.items.all = all;
         }
       }),
+      this.keyNavigation.nextCellCoordsEmitter.subscribe(cellCoords => {
+        if (!this.virtualScroll) {
+          return;
+        }
+
+        if (!cellCoords?.ariaRowIndex) {
+          this.virtualScroll.keyNavItemCoords = null;
+          return;
+        }
+
+        if (cellCoords.ariaRowIndex === this.virtualScroll?.keyNavItemCoords?.ariaRowIndex) {
+          this.virtualScroll.keyNavItemCoords = cellCoords;
+          return;
+        }
+
+        // aria-rowindex is always + 1. Check virtual scroller updateAriaRowIndexes method.
+        const rowIndex = Number(cellCoords.ariaRowIndex) - 1;
```

Comment:
If you apply my other suggestions in the interface.
```suggestion
        const rowIndex = cellCoords.rowIndex - 1;
```